module-apps/apps-common/widgets/DateSetSpinner.cpp

```diff
@@ -0,0 +1,267 @@
+            setMaxDay();
+            applySizeRestrictions();
+            return ret;
+        }
+
+        if (inputEvent.isShortRelease()) {
+            switch (inputEvent.getKeyCode()) {
+            case KeyCode::KEY_ENTER:
+                return handleEnterKey();
+            case KeyCode::KEY_RF:
+                return handleRightFunctionKey();
+            default:
+                break;
+            }
+        }
+        return false;
+    }
+
+    void DateSetSpinner::setMaxDay()
+    {
+        auto dayCoutInMounth =
+            static_cast<unsigned>((date::year(year->get_value()) / date::month(month->get_value()) / date::last).day());
+        setDayMaxRange(dayCoutInMounth);
+
+        if (day->get_value() > dayCoutInMounth) {
+            day->set_value(dayCoutInMounth);
+        }
+    }
+
+    void DateSetSpinner::setDayMaxRange(const std::uint32_t max)
+    {
+        day->set_range_updateless(
+            U8IntegerSpinner::range{static_cast<std::uint8_t>(dayMin), static_cast<std::uint8_t>(max), step});
+    }
+
+    bool DateSetSpinner::handleEnterKey()
+    {
+        if (focusItem == day) {
+            updateFocus(month);
+            return true;
+        }
+        if (focusItem == month) {
+            updateFocus(year);
+            return true;
+        }
+        return false;
+    }
+
+    bool DateSetSpinner::handleRightFunctionKey()
+    {
+        if (focusItem == month) {
+            updateFocus(day);
+            return true;
+        }
+        if (focusItem == year) {
+            updateFocus(month);
+            return true;
+        }
+        return false;
+    }
+
+    void DateSetSpinner::updateFocus(Item *newFocus)
+    {
+        if (focus) {
+            setFocusItem(newFocus);
+            lastFocus = newFocus;
+
+            if (month->focus) {
+                title->setRichText(utils::translate("app_settings_title_month"));
+                updateFont(day, noFocusFontName);
+                updateFont(month, focusFontName);
+                updateFont(year, noFocusFontName);
+            }
+            else if (day->focus) {
+                title->setRichText(utils::translate("app_settings_title_day"));
+                updateFont(day, focusFontName);
+                updateFont(month, noFocusFontName);
+                updateFont(year, noFocusFontName);
+            }
+            else if (year->focus) {
+                title->setRichText(utils::translate("app_settings_title_year"));
+                updateFont(day, noFocusFontName);
+                updateFont(month, noFocusFontName);
+                updateFont(year, focusFontName);
+            }
+        }
+        else {
+            title->setRichText(utils::translate("app_settings_title_day"));
+            updateFont(day, noFocusFontName);
+            updateFont(month, noFocusFontName);
+            updateFont(year, noFocusFontName);
+            setFocusItem(nullptr);
+        }
+
+        applySizeRestrictions();
+    }
```

Comment:
I believe we can do this better than with this bunch of `if`'s
![image](https://user-images.githubusercontent.com/23299649/185403789-d916b2c6-6a22-4560-aca8-3e022064190e.png)

Reply:
This code is copied from TimeSetSpinner. The meme is very funny but could you please be more precisely what would you change here ? Whats wrong in your opinion, the nested If level? code repetition? Would  swith-case for example better here?

Reply:
I did some refactor, please chceck if its better now

Reply:
I did some further refactor based on Onufry's work.